lib/gui/components/safe-webview.js

```diff
@@ -0,0 +1,241 @@
+    return react.createElement('webview', {
+      ref: 'webview',
+      style: {
+        flex: this.state.shouldShow ? null : '0 1',
+        width: this.state.shouldShow ? null : '0',
+        height: this.state.shouldShow ? null : '0'
+      }
+    }, []);
+  }
+
+  /**
+   * @summary Add the Webview events
+   */
+  componentDidMount() {
+
+    // Events React is unaware of have to be handled manually
+    _.map(this.eventTuples, (tuple) => {
+      this.refs.webview.addEventListener(...tuple);
+    });
+
+    // Use the 'success-banner' session
+    this.refs.webview.partition = ELECTRON_SESSION;
+
+    // It's important that this comes after the partition setting, otherwise it will
+    // use another session and we can't change it without destroying the element again
+    this.refs.webview.src = this.entryHref;
+  }
+
+  /**
+   * @summary Remove the Webview events
+   */
+  componentWillUnmount() {
+
+    // Events that React is unaware of have to be handled manually
+    _.map(this.eventTuples, (tuple) => {
+      this.refs.webview.removeEventListener(...tuple);
+    });
+  }
+
+  /**
+   * @summary Refresh the webview if we are navigating away from the success page
+   * @param {Object} nextProps - upcoming properties
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.refreshNow && !this.props.refreshNow) {
+
+      // Reload the page if it hasn't changed, otherwise reset the source URL,
+      // because reload interferes with 'src' setting, resetting the 'src' attribute
+      // to what it was was just prior.
+      if (this.refs.webview.src === this.entryHref) {
+        this.refs.webview.reload();
+
+      } else {
+        this.refs.webview.src = this.entryHref;
+      }
+
+      this.setState({
+        shouldShow: true
+      });
+    }
+  }
+
+  /**
+   * @summary Set the element state to hidden
+   */
+  didFailLoad() {
+    this.setState({
+      shouldShow: false
+    });
+  }
+
+  /**
+   * @summary Set the element state depending on the HTTP response code
+   * @param {Event} event - Event object
+   */
+  didGetResponseDetails(event) {
+    const HTTP_OK = 200;
+    const HTTP_ERR = 400;
+
+    this.setState({
+      shouldShow: event.httpResponseCode >= HTTP_OK && event.httpResponseCode < HTTP_ERR
+    });
+  }
+
+  /**
+   * @summary Open link in browser if it's opened as a 'foreground-tab'
+   * @param {Event} event - event object
+   */
+  static newWindow(event) {
+    const url = new URL(event.url);
+
+    if (_.every([
+      url.protocol === 'http:' || url.protocol === 'https:',
+      event.disposition === 'foreground-tab'
+    ])) {
+      electron.shell.openExternal(url.href);
+    }
+  }
+
+  /**
+   * @summary Forward console messages from the webview to analytics module
+   * @param {Event} event - event object
+   *
+   * @example
+   * console.log({
+   *   type: 0,
+   *   message: 'Hello, World!'
+   * });
+   */
+  static consoleMessage(event) {
+    const LOG_LEVEL = 0;
```

Comment:
Can we re-use our robot existing commands? See https://github.com/resin-io/etcher/blob/master/lib/shared/robot/index.js#L27. We only have "error" and "log" at the moment, but we could also add a "warning" one.